Examples/en/TypeScript/Language/Soundness.ts

```diff
@@ -0,0 +1,110 @@
+// Without a background in type theory, you're unlikely
+// to be familiar with the idea of a type system being "sound"
+
+// Soundness is the idea that the compiler can make guarantees 
+// about the type a value has a runtime, and not just
+// during compilation. This is normal for most programming
+// languages are built with types from day one.
+
+// Building a type system which models a language which has 
+// existed for a few decades however becomes about making 
+// decisions with trade-offs on three qualities: Simplicity,
+// Usability and Soundness.
+
+// With TypeScript's goal of being able to support all JavaScript
+// code, the language tends towards simplicity and usability
+// when presented with ways to add types to JavaScript.
+
+// Let's look at a few cases where TypeScript is provably
+// not sound, to understand what those trade-offs would look
+// like otherwise.
+
+// Type Assertions
+
+const usersAge = ("23" as any) as number;
+
+// TypeScript will let you use type assertions to override
+// the inference to something which is is quite wrong. Using
+// type assertions is a way of telling TypeScript you know
+// best, and TypeScript will try to let you get on with it.
+
+// TODO: Most type systems actually have some sort of runtiem checks
+
+// Function Parameter Bi-variance
+
+// Params for a function support redefining the parameter
+// to be a subtype of the original declaration.
+
+interface InputEvent {
+  timestamp: number;
+}
+interface MouseInputEvent extends InputEvent {
+  x: number;
+  y: number;
+}
+interface KeyboardInputEvent extends InputEvent {
+  keyCode: number;
+}
+
+function listenForEvent(eventType: "keyboard" | "mouse", handler: (event: InputEvent) => void) {}
+
+// You can re-declare the parameter type to be a subtype of
+// the declaration. Above, handler expected a type InputEvent
+// but in the below usage examples - TypeScript accepts
+// a type which has additional properties.
+
+listenForEvent("keyboard", (event: KeyboardInputEvent) => {});
```

Comment:
It might be worth mentioning that this unsoundness is correctly detected using `strictFunctionTypes`, or having a reference to why this is unsound (linking to something on strictness settings).

Reply:
Added, and updated the handbook with this  - thanks